spark/src/main/scala/geotrellis/spark/knn/Implicits.scala

```diff
@@ -0,0 +1,10 @@
+package geotrellis.spark.knn
+
+import geotrellis.vector.Point
+import org.apache.spark.rdd.RDD
+
+object Implicits extends Implicits
+
+trait Implicits {
+  implicit class withKNearestMethods[T](val self: RDD[T]) extends KNearestMethods[T]
```

Comment:
`T` is a bit generic. I can see, because we're saying, hey if you can give us a function that gives an extent then great, we can use this function...so perhaps that should stand. Unfortunate for compile times that this implicit class will be checked for every single call.

I think we should make implicit methods for each of the known cases. E.g. I want to be able to call

```scala
(rdd: RDD[PointFeature[D]]).kNearestNeighbors(point, 5)
(rdd: RDD[Point]).kNearestNeighbors(point, 5)
// etc...
```

Also, we wanted an overload to call for a sequence of points (k nearest neighbors to the cluster centers):

```scala
val clusterCenters: Traversable[Point] = ???
val pointFeatures: RDD[PointFeature[T]] = ???
val result: Map[Point, Seq[PointFeature[T]]] = pointFeatures.kNearestNeighbors(clusterCenters, 5)

// for each key in result, there is a Seq of at most 5 points, which are the 5 closest neighbors to that cluster center
```